### DISCUSSION TITLE
Get augmented data

### DISCUSSION THREAD

**Discussion by qeterme** [explosion/spaCy#12430] [Help: Coding & Implementations] [answered]
Hi,
I'm trying to augment my data during training, but sometimes I would like to see the augmented data. How can I do that?

**Comment by danieldk** [comment, marked as answer]
Thanks for your question! Since you posted this in the _Coding & Implementations_ section, I assume that you'd like to use the augmenters from Python for further exploration. The augmenters themselves are functions that take a `Language` object and an [`Example`](https://spacy.io/api/example) and `yield` an augmented `Example`.

The best way to get an augmenter is to retrieve it by its registered name through the spaCy registry. For example, to get the [`spacy.lower_case.v1`](https://spacy.io/api/top-level#lower_case), you could use

```python
registry.get("augmenters", "spacy.lower_case.v1")
```

This returns a function that constructs the augmenter when given the options specified for the augmenter. Since the lower case augmenter only has the `level` option, we should specify that. E.g.:

```
augmenter = registry.get("augmenters", "spacy.lower_case.v1")(level=0.5)
```

`augmenter` is now a function that you can provide with a `Language` object and an `Example`. Here is a small self-contained example:

```
import spacy
from spacy.util import registry
from spacy.training import Example

# The spaCy registry lets us get the augmenter's builder function. We pass it the
# level argument to construct the augmenter.
augmenter = registry.get("augmenters", "spacy.lower_case.v1")(level=0.5)

nlp = spacy.blank("en")

doc = nlp.make_doc("Let's Augment This Sentence.")
example = Example(doc, doc)

augmented_example = next(augmenter(nlp, example))
# With level=0.5, there is a 50% chance that the example is lowercased.
print(augmented_example.predicted.text)
```
